tools/generate-sponsors.js

```diff
@@ -0,0 +1,140 @@
+/**
+ * @fileoverview
+ * This script generates a markdown file containing sponsor information.
+ * It fetches sponsor data from specified JSON files, formats
+ * the data into HTML sections, and writes the output to a new markdown file.
+ *
+ * Usage:
+ *   Run the script using Node.js:
+ *   node tools/generate-sponsors.js
+ *
+ * Author: Amaresh S M
+ */
+"use strict";
+
+//-----------------------------------------------------------------------------
+// Requirements
+//-----------------------------------------------------------------------------
+
+const fs = require("node:fs");
+const path = require("node:path");
+const { stripIndents } = require("common-tags");
+const got = require("got");
+
+//-----------------------------------------------------------------------------
+// Data
+//-----------------------------------------------------------------------------
+
+const SPONSORS_URL =
+	"https://raw.githubusercontent.com/eslint/eslint.org/main/src/_data/sponsors.json";
+const NEW_FILE_PATH = path.resolve(__dirname, "../sponsors/sponsors.md");
+const TECH_SPONSORS_URL =
+	"https://raw.githubusercontent.com/eslint/eslint.org/main/src/_data/techsponsors.json";
```

Comment:
`src/_data/sponsors.json` and `src/_data/techsponsors.json` are local files here, there's no need to fetch them.